R/utils.R

```diff
@@ -110,3 +110,39 @@
     colData(x) <- cbind(colData(x)[!f], values)
     x
 }
+
+# keep dimnames of feature table (assay) consistent with the meta data 
+# of sample (colData) and feature (rowData)
+.set_feature_tab_dimnames <- function(feature_tab, 
+                                      sample_meta, 
+                                      feature_meta) {
+    if (ncol(sample_meta) > 0) { # sampleMetaFile is not NULL
+        if (ncol(feature_tab) != nrow(sample_meta) 
+            || !setequal(colnames(feature_tab), rownames(sample_meta))) {
```

Comment:
I'm wondering, what happens when sample meta contains only sample IDs without any information? Then sample meta data does not enter to this check, even though sample IDs might be in different order

Should that first if be `if( nrow(sample_meta) > 0 )`?

Reply:
Sorry, I don't understand extractly.

`ncol(sample_meta) = 1` (ncol(sample_meta)  > 0`) when sample meta contains only sample IDs, is that right?

Reply:
I'm thinking if this kind of case is possible

```
> example <- DataFrame(row.names = c("a", "b", "c"))
> ncol(example)
[1] 0
> rownames(example)
[1] "a" "b" "c"
```

Reply:
sample IDs are stored but there are no columns

Reply:
Thanks, got it, may be  like this

`if (nrow(sample_meta) > 0 && ncol(sample_meta) > 0)`